chain/chain/src/chain.rs

```diff
@@ -2466,6 +2481,24 @@ impl Chain {
     }
 }
 
+/// Sandbox node specific operations
+#[cfg(feature = "sandbox")]
+impl Chain {
+    pub fn patch_state(&mut self, records: Vec<StateRecord>) {
+        match self.pending_states_to_patch.take() {
+            None => self.pending_states_to_patch = Some(records),
+            Some(mut pending) => {
+                pending.extend_from_slice(&records);
+                self.pending_states_to_patch = Some(pending);
+            }
+        }
+    }
+
+    pub fn patch_state_finished(&self) -> bool {
```

Comment:
```
    pub fn patch_state_in_progress(&self) -> bool {
        self.pending_states_to_patch.is_some()
    }
```

`finished` is a slight misnomer -- the function returns `true` even if patch state had never been scheduled in the first place.